src/workloads/issues/ConnectionPoolStress.yml

```diff
@@ -0,0 +1,125 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/product-perf"
+Description: |
+  This workload was created to reproduce issues discovered during sharded stress testing. The workload can cause
+  low performance, spikes in latency and excessive connection creation (on mongos processes). The metrics to look at
+  for this workload are the latencies and more specifically the Latency90thPercentile to Latency99thPercentile metrics.
+
+Keywords:
+- reproducer
+- connections
+- secondaryPreferred
+- sharding
+- latency
+
+Parameters:
+- ConnectionThreads: &GlobalConnectionThreads 8000
+- ConnectionDuration: &GlobalConnectionDuration 600 seconds
+- ShardingTaskExecutorPoolMaxSize: &GlobalShardingTaskExecutorPoolMaxSize 32767  # Essentially a noop as this is the default.
+- ShardingTaskExecutorPoolMinSize: &GlobalShardingTaskExecutorPoolMinSize     1  # Essentially a noop as this is the default.
+- ReadMode: &GlobalReadMode secondaryPreferred
+- ClientPerThread: &GlobalSetClientPerThread false
+- MaxPoolSize: &GlobalMaxPoolSize 15000
+
+Constants:
+- &MaxPhases 1
+- Database: &Database test
+- Collection: &Collection Collection0
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: *GlobalMaxPoolSize
+  PerThread:
+    QueryOptions:
+      maxPoolSize: *GlobalMaxPoolSize
+
+Actors:
+- Name: InsertData
+  Type: CrudActor
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Database: *Database
+        Collection: *Collection
+        Operations:
+        - OperationName: updateOne
+          OperationCommand:
+            Filter: {_id: 0}
+            Update: {}
+            OperationOptions: {Upsert: true}
```

Comment:
Oh this is clever, allowing you to run this phase multiple times without having to drop the collection. That might be worth a comment inline.